pom.xml

```diff
@@ -224,6 +225,21 @@ ${git.signoff}
           </preparationGoals>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-javadoc-plugin</artifactId>
+        <version>${maven-javadoc.version}</version>
+        <configuration>
+          <javadocExecutable>${java.home}/bin/javadoc</javadocExecutable>
+          <failOnWarnings>true</failOnWarnings>
+        </configuration>
+        <executions>
+          <execution>
+            <phase>install</phase>
```

Comment:
Note that this phase setting allows `mvn clean install` to pass on javadoc warnings, which allows developers to take care of those warnings at a later time (when they run `mvn javadoc:javadoc`)